scipy/linalg/matfuncs.py

```diff
@@ -668,3 +672,191 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def is_matrix_symmetric(A, rtol=1e-05, atol=1e-08):
+    """
+    Returns True if a matrix is symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    rtol : float
+        Relative tolerance.
+    atol : float
+        Absolute tolerance.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is symmetric.
+
+    Examples
+    --------
+    from scipy.linalg import is_matrix_symmetric
+    m = np.array([[2,3],[3,2]])
+    print(is_matrix_symmetric(m))
+    m = np.array([[5,0],[1,3]])
+    print(is_matrix_symmetric(m))
+    References
+    ----------
+    Hazewinkel, Michiel, ed. (2001) [1994], "Symmetric matrix", Encyclopedia of Mathematics, 
+    Springer Science+Business Media B.V. / Kluwer Academic Publishers, ISBN 978-1-55608-010-4
+    credit to: Nils Werner
+    https://stackoverflow.com/questions/42908334/checking-if-a-matrix-is-symmetric-in-numpy
+    """
+    A = _asarray_square(A)
+    return np.allclose(A, A.T, rtol=rtol, atol=atol)
```

Comment:
This would fail with big and small entries arrays. It needs to be tied to the 1- and inf-norm of the array entries.

Reply:
Replaced with new method.